### ISSUE TITLE
Enigma is broken Linux 32 bits

### ISSUE THREAD

**Issue by rcobraone** [enigma-dev/enigma-dev#1557]
hi,a im testing my old netbook (eee pc 900hd) in linux mint 17.3 ,enigma worked and build, game run is broken.
I used the example from the test one sprite,last update enigma commits 4.333 on 64 bits this error does not occur

check log on GDB and valgrind 
[enigmabug1.txt](https://github.com/enigma-dev/enigma-dev/files/2915508/enigmabug1.txt)
[valgrind-out.txt](https://github.com/enigma-dev/enigma-dev/files/2915510/valgrind-out.txt)

**Comment by RobertBColton** [comment]
It's failing to prepare a vertex buffer basically, the lowest level of our graphics now. Are all games doing this or just a particular game? If it's all games, it could be your driver. I did add a vertex array fallback for vertex buffers in our GL1, but it's possible it's not being enabled. You could try setting it manually in the source and see if that makes a difference.
https://github.com/enigma-dev/enigma-dev/blob/3dd0326edadee06e2895fe3245c338be62da0c57/ENIGMAsystem/SHELL/Graphics_Systems/OpenGL1/GLvertex.cpp#L30

**Comment by RobertBColton** [comment]
I have sent the above proposal as #1558 in the form of a pull request.

**Comment by rcobraone** [comment]
at the beginning all games crash, now it works but there is a problem with displaying or crash

I made a test with my croky game when I run it is ok, if he wants to play, then there is a crash.

my logs
[croky-dbg.txt](https://github.com/enigma-dev/enigma-dev/files/2916835/croky-dbg.txt)
[croky-valgrind-out.txt](https://github.com/enigma-dev/enigma-dev/files/2916841/croky-valgrind-out.txt)

I checked old commits 4.065 here works

**Comment by RobertBColton** [comment]
Ok, because that worked for you, I decided to merge #1558. It seems the rest of your issues are going to stem from graphics problems too. You should check your driver or run the glew test. It seems to be crashing now on index buffers. What would help me is if you could run the game in ENIGMA's debug mode and then obtain the crash log.
```cpp
#1  0x08096531 in enigma_user::index_submit_range (buffer=0, vertex=0, 
    primitive=4, start=0, count=2796)
    at Graphics_Systems/OpenGL1/GLvertex.cpp:276
```
https://github.com/enigma-dev/enigma-dev/blob/078bd7ae9ec55bc2c7f11352b0c0d9ba9bb958b6/ENIGMAsystem/SHELL/Graphics_Systems/OpenGL1/GLvertex.cpp#L276

**Comment by RobertBColton** [comment]
Here is #1559 which is me fixing the fallback for index buffers that are used by tiles (what is reported as not being supported by your graphics card). I just made a minor mistake and was unbinding an index buffer object even when an array was being used for the index buffer.

**Comment by time-killer-games** [comment]
Fuck you ENIGMA. Fuck you and yo daddy.

**Comment by RobertBColton** [comment]
@rcobraone you come back now?

**Comment by rcobraone** [comment]
unfortunately, the fixes do not work as it should change in GLVertex.cpp to bool vbo_is_supported = false, it helps but does not solve the problem
It is possible that this restriction is opengl1.4

**Comment by RobertBColton** [comment]
@rcobraone No, I sent another fix #1559, please check it out, thanks.

**Comment by RobertBColton** [comment]
@rcobraone I implore you to try the second fix, I am 100% certain I know what's going on and the 2nd fix will fix the rest of it.

**Comment by rcobraone** [comment]
not worked, is checked fps engine
[fps_dbg.txt](https://github.com/enigma-dev/enigma-dev/files/2920204/fps_dbg.txt)


my test old version enigma commits 4.065 - X11/Win32: Use fewer globals for window state, game worked

**Comment by RobertBColton** [comment]
And you tested the second fix with the first one in combination? The fps game does not use index buffers, hence you actually need to test both fixes in combination.

**Comment by rcobraone** [comment]
yes, the reason is ARB_vertex_buffer_object support missing in opengl1.4

**Comment by RobertBColton** [comment]
Right, exactly, that's why I created the fallback. But if this is true, then `bool vbo_is_supported = false` should fix the fps game for you. The index buffer fix just has to do with tiles, which is what the other game had a problem with. To fully fix my fallback you need both workarounds at the same time. Finally, I also must add that you are confusing the problem by jumping around to different games. 

I have now merged #1559 so both of my fixes are in master now. I am that certain that they should work.

**Comment by RobertBColton** [comment]
I want to add also that I am sorry we are having these communication problems, I am trying to help. 👍

**Comment by rcobraone** [comment]
ok, I found a solution in GLvertex.cpp should be changed

glGenBuffers, glBindBuffer, glBufferData, glDeleteBuffers

on

glGenBuffersARB, glBindBufferARB, glBufferDataARB ,glDeleteBuffersARB

the solution was here https://en.wikipedia.org/wiki/Vertex_buffer_object

**Comment by RobertBColton** [comment]
Ok, so I see an additional part of the confusion thanks to you. When `GLEW_ARB_vertex_buffer_object` is true, but the GL is older than 1.5, it must be expecting us to use the functions with the ARB in the name. I'll have to ask Josh what he wants to do about this scenario.

**Comment by rcobraone** [comment]
I am in the process of checking the games and examples of whether it works, so far it is ok and works

**Comment by RobertBColton** [comment]
Ok, what you are testing is correct. I'll formalize it in a third final pull request and have Josh review it. I have a reference directly from the GLEW people themselves that say the workaround is acceptable.

**Comment by RobertBColton** [comment]
Alright, I have sent #1560 to fix the remaining issues with the fallback. The fallback should work perfectly now and there will not be any additional issues.

**Comment by rcobraone** [comment]
In most cases, yes, I see a problem with the 3d display.
I think it can be closed

**Comment by RobertBColton** [comment]
Ok, thank you. I will request Josh's review on #1560 when it passes. After I merge it, I will proceed to close this issue. Thank you for all of your help and sticking with me here. I am glad that, despite the communication problems, we seem to have worked out the issue completely. 👍

**Comment by rcobraone** [comment]
thank you, I will try to describe my problems as best as possible

**Comment by RobertBColton** [comment]
Closing as finally resolved by #1560 which I just merged a few moments ago. If you run into other areas where we can improve compatibility please be sure to let us know and share more logs. One of the main reasons I have worked on and added so many graphics systems to ENIGMA is to provide compatibility with the most hardware, so it is something I care about. Cheers! 😸
